src/include/duckdb/storage/single_file_block_manager.hpp

```diff
@@ -49,10 +61,10 @@ class SingleFileBlockManager : public BlockManager {
 
 	FileOpenFlags GetFileFlags(bool create_new) const;
 	//! Creates a new database.
-	void CreateNewDatabase(optional_ptr<ClientContext> context);
+	void CreateNewDatabase(optional_ptr<ClientContext> context, optional_ptr<string> encryption_key = nullptr);
 	//! Loads an existing database. We pass the provided block allocation size as a parameter
 	//! to detect inconsistencies with the file header.
-	void LoadExistingDatabase();
+	void LoadExistingDatabase(optional_ptr<string> encryption_key = nullptr);
```

Comment:
Should this be a `string`?

I also wonder - do we need to pass this into these individual methods? Can't we pass it in using the `StorageManagerOptions` in the constructor?